File: backend/src/main/java/com/whiteblue/backend/domain/user/DTO/GetUserResponse.java
package com.whiteblue.backend.domain.user.DTO;

import lombok.*;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class GetUserResponse {
    @NotBlank
    @Email
    private String username;

    @NotBlank
    private String name;

    private String image;

    @Builder
    public GetUserResponse(String username, String name, String image) {
        this.username = username;
        this.name = name;
        this.image = image;
    }
}

File: backend/src/main/kotlin/com/whiteblue/backend/domain/user/dto/GetUserResponse.kt
package com.whiteblue.backend.domain.user.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

data class GetUserResponse(
    @field:NotBlank
    @field:Email
    val username: String,

    @field:NotBlank
    val name: String,

    val image: String?
)
